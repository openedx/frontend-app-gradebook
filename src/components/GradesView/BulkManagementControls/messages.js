import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  downloadGradesBtn: {
    id: 'gradebook.GradesView.BulkManagementControls.bulkManagementLabel',
    defaultMessage: 'Download Grades',
    description: 'A labeled button that allows an admin user to download course grades all at once (in bulk).',
  },
});

export default messages;
