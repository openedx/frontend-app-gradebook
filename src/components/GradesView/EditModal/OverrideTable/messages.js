import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  adjustedGradeHeader: {
    id: 'gradebook.GradesView.EditModal.Overrides.adjustedGradeHeader',
    defaultMessage: 'Adjusted grade',
    description: 'Edit Modal Override Table Adjusted grade column header',
  },
  dateHeader: {
    id: 'gradebook.GradesView.EditModal.Overrides.dateHeader',
    defaultMessage: 'Date',
    description: 'Edit Modal Override Table Date column header',
  },
  graderHeader: {
    id: 'gradebook.GradesView.EditModal.Overrides.graderHeader',
    defaultMessage: 'Grader',
    description: 'Edit Modal Override Table Grader column header',
  },
  reasonHeader: {
    id: 'gradebook.GradesView.EditModal.Overrides.reasonHeader',
    defaultMessage: 'Reason',
    description: 'Edit Modal Override Table Reason column header',
  },
});

export default messages;
