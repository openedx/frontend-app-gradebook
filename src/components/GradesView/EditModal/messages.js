import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  assignmentHeader: {
    id: 'gradebook.GradesView.EditModal.headers.assignment',
    defaultMessage: 'Assignment',
    description: 'Edit Modal Assignment header',
  },
  currentGradeHeader: {
    id: 'gradebook.GradesView.EditModal.headers.currentGrade',
    defaultMessage: 'Current Grade',
    description: 'Edit Modal Current Grade header',
  },
  originalGradeHeader: {
    id: 'gradebook.GradesView.EditModal.headers.originalGrade',
    defaultMessage: 'Original Grade',
    description: 'Edit Modal Original Grade header',
  },
  studentHeader: {
    id: 'gradebook.GradesView.EditModal.headers.student',
    defaultMessage: 'Student',
    description: 'Edit Modal Student header',
  },
  title: {
    id: 'gradebook.GradesView.EditModal.title',
    defaultMessage: 'Edit Grades',
    description: 'Edit Modal title',
  },
  closeText: {
    id: 'gradebook.GradesView.EditModal.closeText',
    defaultMessage: 'Cancel',
    description: 'Edit Modal close button text',
  },
  visibility: {
    id: 'gradebook.GradesView.EditModal.contactSupport',
    defaultMessage: 'Showing most recent actions (max 5). To see more, please contact support',
    description: 'Edit Modal visibility hint message',
  },
  saveVisibility: {
    id: 'gradebook.GradesView.EditModal.saveVisibility',
    defaultMessage: 'Note: Once you save, your changes will be visible to students.',
    description: 'Edit Modal saved changes effect hint',
  },
  saveGrade: {
    id: 'gradebook.GradesView.EditModal.saveGrade',
    defaultMessage: 'Save Grades',
    description: 'Edit Modal Save button label',
  },
});

export default messages;
