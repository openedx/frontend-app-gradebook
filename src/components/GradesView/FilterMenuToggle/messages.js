import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  editFilters: {
    id: 'gradebook.GradesView.editFilterLabel',
    defaultMessage: 'Edit Filters',
    description: 'A labeled button in the Grades tab that opens/closes the Filters tab, allowing the grades to be filtered',
  },
});

export default messages;
