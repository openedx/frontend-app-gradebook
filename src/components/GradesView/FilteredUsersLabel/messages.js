import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  visibilityLabel: {
    id: 'gradebook.GradesTab.usersVisibilityLabel',
    defaultMessage: 'Showing {filteredUsers} of {totalUsers} total learners',
    description: 'Users visibility label',
  },
});

export default messages;
