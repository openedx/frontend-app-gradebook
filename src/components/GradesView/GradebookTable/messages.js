import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  emailHeading: {
    id: 'gradebook.GradesView.table.headings.email',
    defaultMessage: 'Email',
    description: 'Gradebook table email column header',
  },
  totalGradeHeading: {
    id: 'gradebook.GradesView.table.headings.totalGrade',
    defaultMessage: 'Total Grade (%)',
    description: 'Gradebook table total grade column header',
  },
  usernameHeading: {
    id: 'gradebook.GradesView.table.headings.username',
    defaultMessage: 'Username',
    description: 'Gradebook table username column header',
  },
  studentKeyLabel: {
    id: 'gradebook.GradesView.table.labels.studentKey',
    defaultMessage: 'Student Key*',
    description: 'Gradebook table Student Key label',
  },
  usernameLabel: {
    id: 'gradebook.GradesView.table.labels.username',
    defaultMessage: 'Username',
    description: 'Gradebook table username label',
  },
  totalGradePercentage: {
    id: 'gradebook.GradesView.table.totalGradePercentage',
    defaultMessage: 'Total Grade values are always displayed as a percentage',
    description: 'Gradebook table message that total grades are displayed in percent format',
  },
  noResultsFound: {
    id: 'gradebook.GradesView.table.noResultsFound',
    defaultMessage: 'No results found',
    description: 'Gradebook table message when no learner results were found',
  },
});

export default messages;
