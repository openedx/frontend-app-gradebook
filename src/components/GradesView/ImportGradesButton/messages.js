import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  csvUploadLabel: {
    id: 'gradebook.BulkManagementHistoryView.csvUploadLabel',
    defaultMessage: 'Upload Grade CSV',
    description: 'A labeled button to upload a CSV containing course grades.',
  },
  importGradesBtnText: {
    id: 'gradebook.GradesView.importGradesBtnText',
    defaultMessage: 'Import Grades',
    description: 'A labeled button to import grades in the BulkManagement Tab File Upload Form',
  },
});

export default messages;
