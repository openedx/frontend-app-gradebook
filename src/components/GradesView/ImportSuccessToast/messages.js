import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  description: {
    id: 'gradebook.GradesView.ImportSuccessToast.description',
    defaultMessage: 'Import Successful! Grades will be updated momentarily.',
    description: 'A message congratulating a successful Import of grades',
  },
  showHistoryViewBtn: {
    id: 'gradebook.GradesView.ImportSuccessToast.showHistoryViewBtn',
    defaultMessage: 'View Activity Log',
    description: 'The text on a button that loads a view of the Bulk Management Activity Log',
  },
});

export default messages;
