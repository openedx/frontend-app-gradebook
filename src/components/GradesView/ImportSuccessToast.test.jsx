import React from 'react';
import { shallow } from 'enzyme';

import selectors from 'data/selectors';
import actions from 'data/actions';
import { views } from 'data/constants/app';

import {
  ImportSuccessToast,
  mapStateToProps,
  mapDispatchToProps,
} from './ImportSuccessToast';
import messages from './ImportSuccessToast.messages';

jest.mock('@edx/paragon', () => ({
  Toast: () => 'Toast',
}));
jest.mock('data/selectors', () => ({
  __esModule: true,
  default: {
    app: {
      showImportSuccessToast: (state) => ({ showImportSuccessToast: state }),
    },
  },
}));
jest.mock('data/actions', () => ({
  __esModule: true,
  default: {
    app: {
      setView: jest.fn(),
      setShow: jest.fn(),
    },
  },
}));

describe('ImportSuccessToast component', () => {
  describe('snapshots', () => {
    let el;
    let props = {
      show: true,
    };
    beforeEach(() => {
      props = {
        ...props,
        intl: { formatMessage: (msg) => msg.defaultMessage },
        setAppView: jest.fn(),
        setShow: jest.fn(),
      };
      el = shallow(<ImportSuccessToast {...props} />);
    });
    test('snapshot', () => {
      el.instance().handleShowHistoryView = jest.fn().mockName('handleShowHistoryView');
      el.instance().onClose = jest.fn().mockName('onClose');
      expect(el).toMatchSnapshot();
    });
    describe('Toast props', () => {
      let toastProps;
      beforeEach(() => {
        toastProps = el.props();
      });
      test('action has translated label and onClick from this.handleShowHistoryView', () => {
        expect(toastProps.action).toEqual({
          label: props.intl.formatMessage(messages.showHistoryViewBtn),
          onClick: el.instance().handleShowHistoryView,
        });
      });
      test('onClose from this.onClose method', () => {
        expect(toastProps.onClose).toEqual(el.instance().onClose);
      });
      test('show from show prop', () => {
        expect(toastProps.show).toEqual(props.show);
        el.setProps({ show: false });
        expect(el.props().show).toEqual(false);
      });
    });
    describe('onClose', () => {
      it('calls props.setShow(false)', () => {
        el.instance().onClose();
        expect(props.setShow).toHaveBeenCalledWith(false);
      });
    });
    describe('handleShowHistoryView', () => {
      it('calls setAppView with views.bulkManagementHistory and this.onClose', () => {
        el.instance().onClose = jest.fn();
        el.instance().handleShowHistoryView();
        expect(props.setAppView).toHaveBeenCalledWith(views.bulkManagementHistory);
        expect(el.instance().onClose).toHaveBeenCalled();
      });
    });
  });
  describe('behavior', () => {
  });
  describe('mapStateToProps', () => {
    const testState = { somewhere: 'over', the: 'rainbow' };
    const mapped = mapStateToProps(testState);
    test('show from app showImportSuccessToast selector', () => {
      expect(mapped.show).toEqual(
        selectors.app.showImportSuccessToast(testState),
      );
    });
  });
  describe('mapDispatchToProps', () => {
    test('setAppView from actions.app.setView', () => {
      expect(mapDispatchToProps.setAppView).toEqual(actions.app.setView);
    });
    test('setShow from actions.setShowImportSuccessToast', () => {
      expect(mapDispatchToProps.setShow).toEqual(actions.app.setShowImportSuccessToast);
    });
  });
});
