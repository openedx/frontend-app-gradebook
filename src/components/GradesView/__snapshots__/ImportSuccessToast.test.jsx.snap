// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImportSuccessToast component snapshots snapshot 1`] = `
<Toast
  action={
    Object {
      "label": "View Activity Log",
      "onClick": [Function],
    }
  }
  onClose={[Function]}
  show={true}
>
  Import Successful! Grades will be updated momentarily.
</Toast>
`;
